sample_consensus/include/pcl/sample_consensus/sac_model_registration.h

```diff
@@ -130,12 +130,11 @@ namespace pcl
       setInputTarget (const PointCloudConstPtr &target)
       {
         target_ = target;
-        indices_tgt_.reset (new Indices);
+        indices_tgt_.reset (new Indices (target->size ()));
         // Cache the size and fill the target indices
-        int target_size = static_cast<int> (target->size ());
-        indices_tgt_->resize (target_size);
+        index_t target_size = static_cast<index_t> (target->size ());
 
-        for (int i = 0; i < target_size; ++i)
+        for (index_t i = 0; i < target_size; ++i)
```

Comment:
Good place to use [`std::iota`](https://en.cppreference.com/w/cpp/algorithm/iota)